File: v1/Demo/demo_get_current_pid.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *


def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)

    current_pid = api.get_current_pid()

    if len(current_pid) != 18:
        logger.print_trace_error("get current_pid failed")
    else:
        logger.print_trace("get current_pid success :", current_pid)




if __name__ == "__main__":
    main()

File: v1/Demo/source/demo_get_current_pid.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    vector<float> current_pid = api.get_current_pid();
    if (current_pid.size() != 18)
    {
        Logger::get_instance()->print_trace_error("get current_pid failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get current_pid success : ";
    for (int i = 0; i < current_pid.size(); ++i) {
        std::cout << current_pid[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}